### PR TITLE
Add import from arbitrary binary blobs

### PR DESCRIPTION
**Pull request by CodingVoid**
The Idea is that one can load an arbitrary binary file and extract the sbom data from it.

### CODE REVIEW

File: uswid/cli.py [hunk 1]
```diff
@@ -272,18 +272,17 @@ def main():
...
             elif fmt == SwidFormat.XML:
                 with open(fn, "rb") as f:
                     identity.import_xml(f.read())
             elif fmt == SwidFormat.INI:
                 with open(fn, "rb") as f:
                     identity.import_ini(f.read().decode())
             else:
-                print("{} extension is not supported".format(fn))
-                sys.exit(1)
+                print("{} has unknown extension, try to find uswid data by searching for MAGIC Value".format(fn))
```
Comment (hughsie) [inline uswid/cli.py:282]: maybe something like `{} has unknown extension, using uSWID`

File: uswid/identity.py [hunk 2]
```diff
@@ -80,13 +80,22 @@ def import_bytes(self, blob: bytes, use_header: bool = False) -> None:
...
+        # find and discard magic GUID
         if use_header:
-            (guid, _, hdrsz, payloadsz) = struct.unpack("<16sBHI", blob[:23])
-            if guid != USWID_HEADER_MAGIC:
-                raise NotSupportedError("header does not have expected magic GUID")
+            # find magic GUID in blob
+            offset = 0
+            found_guid = False
+            for offset in range(0, len(blob)):
+                if blob[offset:offset+16] == USWID_HEADER_MAGIC:
```
Comment (hughsie) [inline uswid/identity.py:89]: I think `blob.find(USWID_HEADER_MAGIC)` would work a lot better here, no?

File: uswid/identity.py [hunk 3]
```diff
@@ -80,13 +80,22 @@ def import_bytes(self, blob: bytes, use_header: bool = False) -> None:
...
+                    found_guid = True
+                    print("Found USWID Magic Header")
```
Comment (hughsie) [inline uswid/identity.py:91]: Maybe `Found uSWID header at offset {}`...

**Comment by CodingVoid** [comment]
Changed it.

**Review by hughsie — APPROVED** [review]

**Pull request merged**